hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/api/TestOMDBInsightSearchEndpoint.java

```diff
@@ -0,0 +1,305 @@
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+
+/**
+ * Test for OMDBInsightSearchEndpoint.
+ */
+public class TestOMDBInsightSearchEndpoint extends AbstractReconSqlDBTest {
+  @TempDir
+  private Path temporaryFolder;
+  private OMMetadataManager omMetadataManager;
+  private ReconOMMetadataManager reconOMMetadataManager;
+  private OMDBInsightSearchEndpoint omdbInsightSearchEndpoint;
+  private Random random = new Random();
+  private Set<Long> generatedIds = new HashSet<>();
+  public TestOMDBInsightSearchEndpoint() {
+    super();
+  }
+
+  private long generateUniqueRandomLong() {
+    long newValue;
+    do {
+      newValue = random.nextLong();
+    } while (generatedIds.contains(newValue));
+
+    generatedIds.add(newValue);
+    return newValue;
+  }
+
+  @BeforeEach
+  public void setUp() throws Exception {
+    omMetadataManager = initializeNewOmMetadataManager(
+        Files.createDirectory(temporaryFolder.resolve(
+            "JunitOmMetadata")).toFile());
+    reconOMMetadataManager = getTestReconOmMetadataManager(omMetadataManager,
+        Files.createDirectory(temporaryFolder.resolve(
+            "JunitOmMetadataTest")).toFile());
+    ReconTestInjector reconTestInjector =
+        new ReconTestInjector.Builder(temporaryFolder.toFile())
+            .withReconSqlDb()
+            .withReconOm(reconOMMetadataManager)
+            .withOmServiceProvider(mock(OzoneManagerServiceProviderImpl.class))
+            .addBinding(OzoneStorageContainerManager.class,
+                ReconStorageContainerManagerFacade.class)
+            .withContainerDB()
+            .addBinding(StorageContainerServiceProvider.class,
+                mock(StorageContainerServiceProviderImpl.class))
+            .addBinding(OMDBInsightEndpoint.class)
+            .addBinding(ContainerHealthSchemaManager.class)
+            .build();
+    omdbInsightSearchEndpoint = reconTestInjector.getInstance(
+        OMDBInsightSearchEndpoint.class);
+  }
+
+  @Test
+  public void testSearchOpenKeys() throws Exception {
+    // Create 3 keys in 'bucketOne' and 2 keys in 'bucketTwo'
+    for (int i = 1; i <= 3; i++) {
+      OmKeyInfo omKeyInfo =
+          getOmKeyInfo("sampleVol", "bucketOne", "key_" + i, true);
+      String keyPath = String.format("/sampleVol/%s/key_%d", "bucketOne", i);
+      reconOMMetadataManager.getOpenKeyTable(getBucketLayout())
+          .put(keyPath, omKeyInfo);
+    }
+
+    for (int i = 1; i <= 2; i++) {
+      OmKeyInfo omKeyInfo =
+          getOmKeyInfo("sampleVol", "bucketTwo", "key_" + i, true);
+      String keyPath = String.format("/sampleVol/%s/key_%d", "bucketTwo", i);
+      reconOMMetadataManager.getOpenKeyTable(getBucketLayout())
+          .put(keyPath, omKeyInfo);
+    }
+
+    // Search for keys in 'bucketOne'
+    String searchPrefixBucketOne = "/sampleVol/bucketOne/";
```

Comment:
Pls add some test cases having `searchKey` excluding volume and bucket in path.

Reply:
I've added a test case for searching at the root level. To search within directories, subdirectories, and subpaths, it's essential to include both the volume and bucket names in the search, as they are crucial for defining the absolute path.